### DISCUSSION TITLE
How to disable for specific paths

### DISCUSSION THREAD

**Discussion by gkiely** [vite-pwa/vite-plugin-pwa#545] [Q&A]
I want to disable the service worker for all `/api` paths. I've tried the following as well as variations but after visiting the root `/` it keeps overriding and returning the html for api paths.

```js
VitePWA({
  injectRegister: 'inline',
  registerType: 'autoUpdate',
  workbox: {
    navigateFallbackDenylist: [/^\/api*/],
    runtimeCaching: [
      {
        urlPattern: ({ url }) => url.pathname.startsWith('/api'),
        handler: 'NetworkOnly',
      },
    ],
  },
}),
```

**Comment by userquin** [comment]
Add the `/api` to `wokbox.navigateFallbackDenylist`

**Reply by jswhisperer** [reply]
@userquin hasn't he done that in his shared snippet?
`navigateFallbackDenylist: [/^\/api*/],`

I'm having the same redirect problems and haven't had luck with navigateFallbackDenylist

**Comment by userquin** [comment]
try using `^\/api\//`

**Reply by jswhisperer** [reply]
Thanks you that was making me lose my hair I read like 8 different threads

incase helps someone else a more complete example... my troubled path is /gist not /api
The pages are completely dynamic based on user input so can't be prebuilt / pre-cached

```js
 workbox: {
      globPatterns: ['**/*'],
      navigateFallbackDenylist: [/^\/gist\//],
      globIgnores: ['**/_worker.js/**/*'],
      runtimeCaching: [
        {
          urlPattern: ({ url }) => {
            return url.hostname === 'api.github.com'
          },
          handler: 'CacheFirst',
          options: {
            cacheName: 'github',
            expiration: {
              maxAgeSeconds: 60 * 60 * 24 * 365,
              maxEntries: 500
            }
          }
        }]
        }
}

**Reply by userquin** [reply]
Beware with opaque responses and the size of those opaque responses in the local cache storage...

Read the following link https://developer.chrome.com/docs/workbox/caching-resources-during-runtime#opaque_responses_and_the_navigatorstorage_api

**Reply by jswhisperer** [reply]
You are referring to the glob match all? 
It was fixing an annoying 
`workbox-4245c4a1.js:1 Uncaught (in promise) non-precached-url: non-precached-url :: [{"url":"/"}]`
but doesn't seem to be anymore, I'll try to dig in
https://gist-hub.pages.dev/gist/a2d02c0922f2bb05e72381f73489f571

**Reply by userquin** [reply]
I mean, you're adding a cache first handler for an external url (api.github.com), every response cached from the api call, can be stored with an extra size of 7MB, read the link

**Reply by jswhisperer** [reply]
Oh interesting, thank you for the link, I'll try to optimise my cache, from what I can see cacheStorage only has the one REST request teh the API 

<img width="847" alt="image" src="https://github.com/user-attachments/assets/49e9f217-8945-4dce-9381-d9b9c58244f7" />

and the size seems to be unpadded 

<img width="440" alt="image" src="https://github.com/user-attachments/assets/f8742039-d91e-4414-b8ec-eb9c18ab48e7" />
